server/s3_get_object_action.cc

```diff
@@ -320,6 +331,13 @@ void S3GetObjectAction::read_object() {
   set_total_blocks_to_read_from_object();
   motr_reader = motr_reader_factory->create_motr_reader(
       request, object_metadata->get_oid(), object_metadata->get_layout_id());
+  size_t part_one_size = object_metadata->get_part_one_size();
+  if (part_one_size > 0) {
+    motr_reader->set_multipart_part_size(part_one_size);
+    std::string etag = object_metadata->get_md5();
+    std::string num_of_parts = etag.substr(etag.rfind('-') + 1);
```

Comment:
This looks like we have some naming convention embedded in the software.  If so, this worries me because it is easy to forget the convention and suddenly break code.  Perhaps there should be helper functions to build/parse this string so that the conventions can be isolated to just a single location?

Reply:
Excellent point. This etag format is created in `server/s3_aws_etag`. I just have refactored this piece of code to that file, so operations with this format doesn't leave a single file. There is still knowledge that the etag embeds # of parts, but know it's a matter of leaky abstractions.